File: k9mail/src/main/java/com/fsck/k9/helper/EmailHelper.java
package com.fsck.k9.helper;


public final class EmailHelper {
    public static String getDomainFromEmailAddress(String email) {
        int separatorIndex = email.lastIndexOf('@');
        if (separatorIndex == -1 || separatorIndex + 1 == email.length()) {
            return null;
        }

        return email.substring(separatorIndex + 1);
    }

    public static String getProviderNameFromEmailAddress(String email) {
        String domain = getDomainFromEmailAddress(email);
        if (domain == null) return null;

        int dotIndex = domain.lastIndexOf(".");
        if (dotIndex == -1) return null;

        return domain.substring(0, dotIndex);
    }

    public static String[] splitEmail(String email) {
        String[] retParts = new String[2];
        String[] emailParts = email.split("@");
        retParts[0] = (emailParts.length > 0) ? emailParts[0] : "";
        retParts[1] = (emailParts.length > 1) ? emailParts[1] : "";
        return retParts;
    }
}

File: mail/common/src/main/java/com/fsck/k9/helper/EmailHelper.kt
package com.fsck.k9.helper

object EmailHelper {
    @JvmStatic
    fun getLocalPartFromEmailAddress(email: String): String? {
        val index = email.lastIndexOf('@')
        return if (index == -1 || index == email.lastIndex) null else email.substring(0, index)
    }

    @JvmStatic
    fun getDomainFromEmailAddress(email: String): String? {
        val index = email.lastIndexOf('@')
        return if (index == -1 || index == email.lastIndex) null else email.substring(index + 1)
    }
}
